client/src/main/scala/skuber/api/WatchSource.scala

```diff
@@ -0,0 +1,121 @@
+package skuber.api
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model._
+import akka.stream.scaladsl.{Broadcast, Flow, GraphDSL, JsonFraming, Merge, Source}
+import akka.stream.{Materializer, SourceShape}
+import akka.util.ByteString
+import play.api.libs.json.{Format, JsError, JsSuccess, Json}
+import skuber.api.client._
+import skuber.{ObjectResource, ResourceDefinition}
+
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor}
+import scala.concurrent.duration._
+
+object WatchSource {
+  private sealed trait StreamElement[O <: ObjectResource] {}
+  private case class End[O <: ObjectResource]() extends StreamElement[O]
+  private case class Request[O <: ObjectResource](resourceVersion: Option[String]) extends StreamElement[O]
```

Comment:
A bit of a quibble but can we rename this to something like `Start` or `Since` - calling it `Request` doesn't seem quite right as it doesn't actually reference the request, just the resource version from which the Watch should start watching.